src/txn/txn_ckpt.c

```diff
@@ -1880,13 +1988,20 @@ __wt_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
       !WT_IS_METADATA(session->dhandle) ||
         FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_METADATA));
 
+    /* Discard the cached checkpoint list when checkpointing a single file by itself. */
+    __wt_meta_saved_ckptlist_free(session);
+
     WT_RET(__wt_config_gets_def(session, cfg, "force", 0, &cval));
     force = cval.val != 0;
     WT_SAVE_DHANDLE(session, ret = __checkpoint_lock_dirty_tree(session, true, force, true, cfg));
-    WT_RET(ret);
-    if (F_ISSET(S2BT(session), WT_BTREE_SKIP_CKPT))
-        return (0);
-    return (__checkpoint_tree(session, true, cfg));
+    if (ret != 0 || F_ISSET(btree, WT_BTREE_SKIP_CKPT))
```

Comment:
You could use `S2BT` here and avoid adding the new local variable for one access.

Reply:
done